src/Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <vector>
+#include <Core/NamesAndTypes.h>
+#include <Core/Range.h>
+
+namespace Iceberg
+{
+
+enum class PartitionTransform
+{
+    Year,
+    Month,
+    Day,
+    Hour,
+    Identity,
+    Void,
+    Unsupported
+};
+
+struct SpecificSchemaPartitionInfo
+{
+    std::vector<std::vector<DB::Range>> ranges;
+    DB::NamesAndTypesList partition_names_and_types;
+};
```

Comment:
Why is it called `Specific`? If I understood correctly, this struct contains information about partitions for all files inside a single manifest file, is it right? Also from field name `ranges` it's not clear what is it and why it's a vector of vector. Please, write comments or use better naming.
Also, as this struct is related only to manifest file, let's move it to the `ManifestFile` class